src/extraction/index.js

```diff
@@ -28,40 +28,72 @@ const EXTRACTION_METHODS = {
   open_graph: extractProductWithOpenGraph,
 };
 
+const IS_BACKGROUND_UPDATE = (function isBackgroundUpdate() {
+  let result = false;
+  try {
+    result = window.top.location.href.startsWith(
+      browser.runtime.getURL('/'), // URL is unique per-install / hard to forge
+    );
+  } catch (err) {
+    // Non-background updates may throw a cross-origin error
+  }
+  return result;
+}());
+
+/**
+ * Helper class to record extraction-related telemetry events.
+ */
+class ExtractionAttempt {
```

Comment:
![](https://3.bp.blogspot.com/-mxRvNXiwXJY/WPBwnUMUWKI/AAAAAAAAIWA/fZWqK3ZXAJgb4LaWMn3YUIJtwxOErReCACLcB/s320/Head%2BFirst%2BAnalaysis%2Band%2BDesign.JPG)

Reply:
![68747470733a2f2f332e62702e626c6f6773706f742e636f6d2f2d6d7852764e586977584a592f575042776e554d55574b492f41414141414141414957412f665a57714b335a58414a6762344c61574d6e335955494a7477784f457252654341434c63422f733332302f4865616425324](https://user-images.githubusercontent.com/23885/48032235-0be5b880-e114-11e8-874b-eca063815363.jpg)

Reply:
That guy looks pretty smug.